### ISSUE TITLE
Unit tests for the HyperTransport

### ISSUE THREAD

**Issue by cmbrandenburg** [chill-rs/chill#16]
The `HyperTransport` is tested indirectly, via the integration tests. Consider writing unit tests for it to ensure all cases are covered.

**Comment by cmbrandenburg** [comment]
@jgillich No, I haven't considered it, but I like the idea. What do you have in mind?

The main hurdle I see is that the Chill transport is tailored for CouchDB and doesn't extend to other applications. For example, see the [`with_revision_query`](https://github.com/chill-rs/chill/blob/a34c4110652c77dd026b4d661f9ef13f1f6c5982/src/transport/mod.rs#L98-L101). That method makes sense only in the context of CouchDB. How do we generalize the transport into a micro-framework useful for other APIs?

**Comment by jgillich** [comment]
I haven't entirely thought this through yet, but revision is a query parameter, right? You could assume standard HTTP types like query parameters and headers, which the transport could then use those to and construct a HTTP/WS/HTCPCP request or parse them right away for testing.

But I have to think about that for while, there probably is a better way.

**Comment by cmbrandenburg** [comment]
@jgillich I'm on board with the general idea. Any code in Chill that isn't fundamental to CouchDB is better off existing in another crate. However, I've got higher priorities and don't foresee working on this anytime soon. Feel free to take ownership of the idea and make it happen!

**Comment by cmbrandenburg** [comment]
@jgillich I opened #51, which, if it works out, might make the transport layer amenable to being separated into its own crate. It remains to be seen how useful such a “micro-framework” would be to other client-side HTTP libraries.
